xblockutils/studio_editable.py

```diff
@@ -0,0 +1,295 @@
+                info['values'] = values
+        if info["type"] in ("list", "set") and field.runtime_options.get('list_values_provider'):
+            list_values = field.runtime_options['list_values_provider'](self)
+            # list_values must be a list of values or {"display_name": x, "value": y} objects
+            # Furthermore, we need to convert all values to JSON since they could be of any type
+            if list_values and (not isinstance(list_values[0], dict) or "display_name" not in list_values[0]):
+                list_values = [json.dumps(val) for val in list_values]
+                list_values = [{"display_name": val, "value": val} for val in list_values]
+            else:
+                for entry in list_values:
+                    entry["value"] = json.dumps(entry["value"])
+            info['list_values'] = list_values
+            info['has_list_values'] = True
+        return info
+
+    @XBlock.json_handler
+    def submit_studio_edits(self, data, suffix=''):
+        """
+        AJAX handler for studio_view() Save button
+        """
+        values = {}  # dict of new field values we are updating
+        to_reset = []  # list of field names to delete from this XBlock
+        for field_name in self.editable_fields:
+            field = self.fields[field_name]
+            if field_name in data['values']:
+                if isinstance(field, JSONField):
+                    values[field_name] = field.from_json(data['values'][field_name])
+                else:
+                    raise JsonHandlerError(400, "Unsupported field type: {}".format(field_name))
+            elif field_name in data['defaults'] and field.is_set_on(self):
+                to_reset.append(field_name)
+        self.clean_studio_edits(values)
+        validation = Validation(self.scope_ids.usage_id)
+        # We cannot set the fields on self yet, because even if validation fails, studio is going to save any changes we
+        # make. So we create a "fake" object that has all the field values we are about to set.
+        preview_data = FutureFields(
+            new_fields_dict=values,
+            newly_removed_fields=to_reset,
+            fallback_obj=self
+        )
+        self.validate_field_data(validation, preview_data)
+        if validation:
+            for field_name, value in values.iteritems():
+                setattr(self, field_name, value)
+            for field_name in to_reset:
+                self.fields[field_name].delete_from(self)
+            return {'result': 'success'}
+        else:
+            raise JsonHandlerError(400, validation.to_json())
+
+    def clean_studio_edits(self, data):
+        """
+        Given POST data dictionary 'data', clean the data before validating it.
+        e.g. fix capitalization, remove trailing spaces, etc.
+        """
+        # Example:
+        # if "name" in data:
+        #     data["name"] = data["name"].strip()
+        pass
```

Comment:
No, because many subclasses may not have any cleaning to do. I want this to be an optional override and not a mandatory one, in order to reduce code clutter.